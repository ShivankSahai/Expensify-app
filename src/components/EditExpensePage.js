import React from 'react';
import {connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import {editExpense,removeExpense} from '../actions/expenses'; 

const EditExpensePage=(props)=>{
    // console.log(props);
    return (
        <div>
            <ExpenseForm 
                expense={props.expense}
                onSubmit={(expense)=>{
                    props.dispatch(editExpense(props.expense.id,expense));
                    
                    console.log(expense);
                }}/>
            <button onClick={()=>{
                props.dispatch(removeExpense({id:props.expense.id}));
                props.history.push('/');
                }}>
                remove
                
            </button>
    
        </div>
    );

};

const mapToState=(state,props)=>{
    return{
        expense: state.expenses.find((expense)=>{
            return expense.id===props.match.params.id;
        })
    };
}
export default connect(mapToState)(EditExpensePage);