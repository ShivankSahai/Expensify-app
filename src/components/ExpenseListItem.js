import React from 'react';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem=({dispatch,description,amount,createdAt,id})=>(
    <div>
        <Link to={'edit/'+id}>
            <h3>{description}</h3>
        </Link>
        
        <p>Rs.{numeral(amount).format('Rs0,0.00')}- {moment(createdAt).format('MMMM D0, YYYY')}</p>
    </div>
);

export default (ExpenseListItem);