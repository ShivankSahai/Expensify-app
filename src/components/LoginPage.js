import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

export const LoginPage=({startLogin})=>(
    
        <div className="box-layout">
            <div className="box-layout__box">
                <h1 className="box-layout__title">EXPENSIFY</h1>
                <h3>Keep a check on your expenses!</h3>
                <button className="button" 
                    onClick={startLogin}>LOGIN WITH GOOGLE</button>
            </div>
        </div>
    )

const mapDispatchToProps=(dispatch)=>({
    startLogin: ()=>dispatch(startLogin())
})

export default connect(undefined,mapDispatchToProps)(LoginPage);
