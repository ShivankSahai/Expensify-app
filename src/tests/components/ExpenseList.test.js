import React from 'react';
import {shallow} from 'enzyme';
import {ExpenseList} from '../../components/ExpenseList';
import expenses from '../fixtures/expenses';

test('shoudl render Expenselist with expenses',()=>{
    const wrapper=shallow(<ExpenseList expenses={expenses}/>)
    expect(wrapper).toMatchSnapshot();
})

test('shoudl renderr expense list with empty message',()=>{
    const wrapper=shallow(<ExpenseList expenses={[]}/>)
    expect(wrapper).toMatchSnapshot();
})
