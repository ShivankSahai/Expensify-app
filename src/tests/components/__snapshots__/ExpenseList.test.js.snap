// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shoudl render Expenselist with expenses 1`] = `
<div>
  <ExpenseListItem
    amount={5}
    createdAt={0}
    description="gum"
    id="1"
    key="1"
    note=""
  />
  <ExpenseListItem
    amount={2000}
    createdAt={-345600000}
    description="rent"
    id="2"
    key="2"
    note=""
  />
  <ExpenseListItem
    amount={5000}
    createdAt={345600000}
    description="credit"
    id="3"
    key="3"
    note=""
  />
</div>
`;

exports[`shoudl renderr expense list with empty message 1`] = `
<div>
  <p>
    No expenses
  </p>
</div>
`;
